### PR TITLE
new(tests): EOF - EIP-6206: Add stack overflow by rule check to JUMPF

### PR DESCRIPTION
**Pull request by shemnon**
Signed-off-by: Danno Ferrin <danno@numisight.com>

## 🗒️ Description

Just like CALLF, JUMPF has a stack overflow by rule check that
tests were not tripping.

## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Review by pdobacz — APPROVED** [review]

File: tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py [hunk 1]
```diff
@@ -235,39 +235,64 @@ def test_jumpf_stack_size_1024_at_push(
...
-                    # Runtime stack overflow
-                    Op.POP + Op.POP + Op.RETF,
+                    Op.CALLDATALOAD(0)
+                    + Op.ISZERO
+                    + Op.RJUMPI[6]
+                    + Op.PUSH0 * 3
+                    + Op.POP * 3
+                    + Op.SSTORE(slot_stack_canary, value_canary_written)
+                    + Op.RETF,
                     code_inputs=0,
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py:284]: Just for completeness sake, since the formula includes the target section inputs, I think we should the number of inputs as a test parameter. Perhaps we just pop the inputs at the start of the code before the `CALLDATALOAD`.

File: tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py [hunk 2]
```diff
@@ -235,39 +235,64 @@ def test_jumpf_stack_size_1024_at_push(
...
+    failure: bool,
     eof_state_test: EOFStateTestFiller,
 ):
-    """Test stack overflowing 1024 items in JUMPF target function"""
+    """
+    Test stack overflowing 1024 items in JUMPF target function
+
+    `no_overflow` - the stack does not overflow at JUMPF call, executes to end
+    `rule_overflow` - reserved stack overflows, but execution would not overflow
+    `execution_overflow` - execution would overflow (but still blocked by reserved stack rule)
```
Comment (marioevz) [inline tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py:256]: I don't quite understand the difference between these two scenarios. What is the reserved stack?

**Review by marioevz — COMMENTED** [review]
Some comments/questions. Thanks!

File: tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py (hunk 2 above)
Comment (shemnon) [reply tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py]: Elaborated in the comments.  Without rule #2 in the EIP then the `rule_overflow` case would execute correctly. `execution_overflow` fails without rule #2 as well, but rule #2 kills it first.  No way to distinguish the two failures in-evm however, so we need to test the "rule overflow" case to ensure it's being applied.

File: tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py (hunk 1 above)
Comment (shemnon) [reply tests/osaka/eip7692_eof_v1/eip6206_jumpf/test_jumpf_execution.py:297]: Added params, and some more test cases to exercise the params.

**Review by marioevz — APPROVED** [review]
LGTM, thanks!
